mycpp/cppgen_pass.py

```diff
@@ -2419,10 +2418,14 @@ def visit_func_def(self, o: 'mypy.nodes.FuncDef') -> T:
                 arg_names = [arg.variable.name for arg in o.arguments]
                 #log('arg_names %s', arg_names)
                 #log('local_vars %s', self.local_vars[o])
-                self.prepend_to_block = [
-                    (lval_name, c_type, lval_name in arg_names)
-                    for (lval_name, c_type) in self.local_vars[o]
-                ]
+                self.prepend_to_block = []
+                for (lval_name, lval_type) in self.local_vars[o]:
+                    c_type = lval_type
+                    if not isinstance(lval_type, str):
```

Comment:
hm can we remove this and always use a Type instance rather than a string?

Reply:
I wanted to do this, but the casting case makes it difficult: https://github.com/oilshell/oil/blob/master/mycpp/cppgen_pass.py#L1482-L1499). We would need some way of knowing whether to call `GetCType()` or `_GetCTypeForCast()` when iterating over the list of locals. Any way of communicating that felt messier than falling back on a string

Reply:
So we have to distinguish the cases

    x = 42
    x = cast(SomeType, y)

How about  adding a boolean field to local_var_list that tells if it's casted or not?

Reply:
Yeah, I had something like this in my first draft of this. It felt kinda clunky. I guess I don't feel too strongly about this, though. I'll revert to that.